app/client/src/utils/PerformanceTracker.ts

```diff
@@ -51,6 +56,7 @@ class PerformanceTracker {
     eventName: PerformanceTransactionName,
     data?: any,
     skipLog = false,
+    tags: null | Array<PerfTag> = null,
```

Comment:
You could just default to an empty array instead. This way you wont have to do the check for this existing below
```suggestion
    tags: Array<PerfTag> = [],
```